### PR TITLE
feat(argo-cd): Add probes for ApplicationSet controller

### PR DESCRIPTION
**Pull request by pdrastil**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
We cannot implement features which are not available upstream.

Ref: https://github.com/argoproj/argo-cd/blob/v2.4.14/manifests/base/applicationset-controller/argocd-applicationset-controller-deployment.yaml

**Comment by pdrastil** [comment]
@mkilchhofer probes are implemented in controller since argolabs - we even have args for that configured

**Comment by mkilchhofer** [comment]
@pdrastil where? Can you provide a link?

**Comment by pdrastil** [comment]
@mkilchhofer - Sure

Controller - https://github.com/argoproj/argo-cd/blob/029be590bfd5003d65ddabb4d4cb8a31bff29c18/cmd/argocd-applicationset-controller/commands/applicationset_controller.go#L200

Chart:

values.yaml - https://github.com/argoproj/argo-helm/blob/3a05c7e7b45c094cec8158fb205cd00983dfb164/charts/argo-cd/values.yaml#L2009
deployment.yaml (wrong name - not a http) - https://github.com/argoproj/argo-helm/blob/3a05c7e7b45c094cec8158fb205cd00983dfb164/charts/argo-cd/templates/argocd-applicationset/deployment.yaml#L61

I've tested this and TCP probe works

File: charts/argo-cd/templates/argocd-applicationset/deployment.yaml [hunk 1]
```diff
@@ -58,15 +58,31 @@ spec:
...
+            successThreshold: {{ .Values.applicationSet.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.applicationSet.livenessProbe.failureThreshold }}
+          readinessProbe:
+            tcpSocket:
+              port: probe
+            initialDelaySeconds: {{ .Values.applicationSet.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.applicationSet.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.applicationSet.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.applicationSet.readinessProbe.successThreshold }}
+            failureThreshold: {{ .Values.applicationSet.readinessProbe.failureThreshold }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-applicationset/deployment.yaml:85]: Not in upstream manifests

Comment (pdrastil) [reply charts/argo-cd/templates/argocd-applicationset/deployment.yaml:85]: how about making this optional same as in Dex?

**Comment by pdrastil** [comment]
@mkilchhofer made the probes optional

**Review by mkilchhofer — APPROVED** [review]
👍

**Pull request merged**